### DISCUSSION TITLE
A problem with the keyboard light

### DISCUSSION THREAD

**Discussion by alpenllebe** [seerge/g-helper#1643] [Q&A]
Starting Asus Optimization will cause the SHIFT+M1 M2 hotkey to become invalid (keyboard light)

Can the keyboard light be set to turn off completely after being turned off? The current situation is that after turning off the keyboard light, the keyboard light will still light up during the boot process. Unless the check box is unchecked in this option, the keyboard light will not light up during the boot process, but this will cause me to need to check this option again before I can turn on the keyboard the next time I need to turn on the keyboard light. lamp.
Can you add this function? Thank you very much to the author.
![屏幕截图 2023-11-24 124600_LI](https://github.com/seerge/g-helper/assets/151899833/d3377924-439f-4b3c-a562-276747226c19)

**Comment by seerge** [comment]
@alpenllebe when app sees that asus optimization service is running, it doesn't control your backlight brigghtness (cause that service will), it's absolutely intentional.

As for backlight status, "awake" from asus hardware's point of view - is when laptop is ON, it also includes time when laptop is booting (i.e. loading windows) or shutting down :) this is how things are

**Reply by alpenllebe** [reply]
Wow, the reply is very fast. Thank you very much to the author. I understand the first one.
Can the second code be set by an advanced user? I want to set it up like that (that is, after I turn off the keyboard light, it will not light up when turning on the computer, waking up, etc.).

**Comment by seerge** [comment]
@alpenllebe i can't change that. It depends on how hardware works.

Most of the laptop models actually remember backlight brightness after reboot, and doesn't light up keyboard if it was off before :)

**Reply by alpenllebe** [reply]
非常感谢作者^_^

**Reply by alpenllebe** [reply]
Thank you very much to the author
